snapshot/containerd/content.go

```diff
@@ -115,7 +115,20 @@ func (c *nsFallbackStore) Update(ctx context.Context, info content.Info, fieldpa
 }
 
 func (c *nsFallbackStore) Walk(ctx context.Context, fn content.WalkFunc, filters ...string) error {
-	return c.main.Walk(ctx, fn, filters...)
+	seen := make(map[digest.Digest]struct{})
```

Comment:
This enables history namespace leak testing for oci workers as well where previously it only happened for containerd worker. So technically it should make the flaky test issue more common if it was not fixed.